src/vm/ast/parser/mod.rs

```diff
@@ -180,11 +180,15 @@ pub fn lex(input: &str) -> ParseResult<Vec<(LexItem, u32, u32)>> {
                         Ok(LexItem::Whitespace)
                     },
                     TokenType::Comma => {
-                        context = LexContext::ExpectNothing;
+                        // comma should not be followed directly by an item,
+                        //  e.g., {a: 1,b: 2} should not be legal
+                        context = LexContext::ExpectClosing;
                         Ok(LexItem::CommaSeparator)
                     },
                     TokenType::Colon => {
-                        context = LexContext::ExpectNothing;
+                        // colon should not be followed directly by an item,
+                        //  e.g., {a:b} should not be legal
```

Comment:
Also confused by this one.  The PR says that trailing commas are legal, but not required.

Reply:
In several lisp variants, `a:b` and `a,b` would be valid identifiers, and I want to avoid any possibility of confusion there.

Reply:
Ah, okay :+1: